test/cql-pytest/test_describe.py

```diff
@@ -552,6 +552,28 @@ def test_whitespaces_in_table_options(cql, test_keyspace, scylla_only):
         desc = "\n".join([d.create_statement for d in cql.execute(f"DESC TABLE {tbl}")])
         assert re.search(regex, desc) == None
 
+# In order to restore schema from `DESC SCHEMA`
+# UDTs have to be sorted topologically because they can depent on each other
+# The test doesn't check validity of UDT description, 
+# but only the order in `DESC KEYSPACE ks`(it's equivalent to part of `DESC SCHEMA`)
+def test_udts_order(scylla_only, cql, test_keyspace):
+    udt_names = [chr(ord('z') - i) + unique_name() for i in range(5)]
+    with new_type(cql, test_keyspace, "(a int)", udt_names[0]) as t0:
+        with new_type(cql, test_keyspace, f"(a frozen<{t0}>)", udt_names[1]) as t1:
+            with new_type(cql, test_keyspace, f"(a frozen<list<{t0}>>, b set<frozen<{t1}>>)", udt_names[2]) as t2:
+                with new_type(cql, test_keyspace, f"(a frozen<tuple<{t0}, {t2}>>)", udt_names[3]) as t3:
+                    with new_type(cql, test_keyspace, f"(a map<frozen<{t1}>, frozen<{t2}>>, b list<frozen<{t0}>>)", udt_names[4]) as t4:
```

Comment:
Instead of deeply nesting the `with` statement, you can use multiple assignments in a single `with` statement (ref: https://docs.python.org/3/reference/compound_stmts.html#the-with-statement):

```python
    with new_type(cql, test_keyspace, "(a int)", udt_names[0]) as t0, \
            new_type(cql, test_keyspace, f"(a frozen<{t0}>)", udt_names[1]) as t1, \
            new_type(cql, test_keyspace, f"(a frozen<list<{t0}>>, b set<frozen<{t1}>>)", udt_names[2]) as t2, \
            new_type(cql, test_keyspace, f"(a frozen<tuple<{t0}, {t2}>>)", udt_names[3]) as t3, \
            new_type(cql, test_keyspace, f"(a map<frozen<{t1}>, frozen<{t2}>>, b list<frozen<{t0}>>)", udt_names[4]) as t4:
```

See also `test_mv_synchronous_updates` in `test_materialized_view.py` for an example use.